imports/plugins/core/orders/server/no-meteor/mutations/placeOrder.js

```diff
@@ -261,6 +272,25 @@ export default async function placeOrder(context, input) {
     }
   };
 
+  let referenceId;
+  const createReferenceIdFunctions = getFunctionsOfType("createOrderReferenceId");
+  if (!createReferenceIdFunctions || createReferenceIdFunctions.length === 0) {
+    // if the cart has a reference Id, and no custom function is created use that
+    if (_.get(cart, "referenceId")) { // we want the else to fallthrough if no cart to keep the if/else logic simple
+      ({ referenceId } = cart);
+    } else {
+      referenceId = Random.id();
+    }
+  } else {
+    referenceId = createReferenceIdFunctions[0](cart, order);
```

Comment:
Can you pass `order` arg before `cart` arg since `cart` may be undefined and some people may not need it?

Reply:
Actually a couple more things:

1. Do `= await createReferenceIdFunctions[0](order, cart)` (add `await`) in case the plugin needs to do async calls to generate an ID.
2. Specifically check `referenceId` after this and if it isn't a string, throw an error with a message something like `"createOrderReferenceId" function returned invalid ID`. The schema check later would catch this, but if that happened, it would be a generic error and would take some investigating to figure out that it's an issue with the registered function.

Do both of these things for the cart ID function, too.